python/pyarrow/_csv.pyx

```diff
@@ -178,6 +183,33 @@ cdef class ReadOptions(_Weakrefable):
         The number of rows to skip before the column names (if any)
         and the CSV data.
         See `skip_rows_after_names` for interaction description
+
+        Examples:
+        ---------
+        >>> from pyarrow import csv
```

Comment:
I'm not sure it's worth repeating the import at the top of each example. But otherwise you should add one at the top of the `use_threads` example.

@jorisvandenbossche Thoughts?

Reply:
We were just talking today with Joris that I should put all the examples under the `csv.read_csv()` and so this will not be repeating.

Reply:
> put all the examples under the `csv.read_csv()` and so this will not be repeating.

All in the `csv.ReadOptions` docstring (and `csv.ConvertOptions` etc), or all under `csv.read_csv` ?

Reply:
I would suggest to add all under `csv.read_csv` as there is more space available (html version) and I would be able to connect between different options/examples.